### PR TITLE
Banking api and implementation

### PR DESCRIPTION
**Pull request by joskuijpers**
This pull request adds a banking abstract API with a Bunq implementation.

Not everything is done yet, but it is possible to make a session and request.
Another feature is the IBANVerifier class.

A couple of end-to-end tests have been made but need an API key for travis.
Travis fails to build this branch, because of configuration issues with the new Java8 configuration.

To merge: fix the travis build. Make a travis API key and commit it.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ before_install:
 
 before_script:
   # Create and start emulator
-  - echo no | android create avd --force -n test -t android-21 --abi armeabi-v7a
+  - echo no | android create avd --force -n test -t android-23 --abi armeabi-v7a
```
Comment (MatthijsKok) [inline .travis.yml:41]: this change breaks travis i think

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/keys/PrivateKey.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
     /**
      * The public key that belongs to this private key.
      */
+    @SuppressWarnings("unused")
```
Comment (MatthijsKok) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/keys/PrivateKey.java]: This class needs to be reverted. It's deleted.

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/Response.java [hunk 3]
```diff
@@ -45,7 +46,8 @@ private void decryptChallenge() {
     /**
      * Verify the signature of the Challenge.
      * @return Whether the signature is valid.
-     * @throws SignatureException If the Challenge signature is invalid. This is a sign of malicious activity.
+     * @throws SignatureException If the Challenge signature is invalid.
+     *   This is a sign of malicious activity.
```
Comment (MatthijsKok) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/Response.java]: This class needs to be reverted. It's changed.

File: .travis.yml (hunk 1 above)
Comment (joskuijpers) [reply .travis.yml:41]: Yeah, but we don't support android-21 :(

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/Response.java (hunk 3 above)
Comment (joskuijpers) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/Response.java]: remove the whole class?

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/keys/PrivateKey.java (hunk 2 above)
Comment (joskuijpers) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/keys/PrivateKey.java]: remove the whole class?

**Pull request merged**